### PR TITLE
feat(windows): Online Update check stream update data 💽

### LINKED ISSUE #10045: feat(windows): Update Tab for Keyman Configuration as per functional spec
Other work:

- [ ] #10117

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #10045
Relates to #10038.

Starts to refactor the online update check in Windows so that the results are stored to disk and can be checked at any time. This commit:
* removes keepintouch frame
* adds update frame
* adds serialization of update check data
* presents the serialized update check data in the update frame

Online updates won't currently work, as more work is needed in follow-up PRs

## Screenshots

When no update is available:

image

When updates are available:

image

Note: the 'SKUpdate_...' text will be replaced with the actual update description in a full build.

## TODO

- [x] Add Keep in Touch link to Support tab
- [x] Cleanup the layout of the Update tab
  - [x] Remove checkmarks for selecting updates to Update tab
  - [x] Add strings for Update tab
  - [x] Icon for the update tab
  - [x] Red circled number for #updates available in the left-hand menu Update item
- [x] Provide actual 'status' text relevant to current cache status

### CODE REVIEW

File: windows/src/desktop/kmshell/kmshell.dpr [hunk 1]
```diff
@@ -198,7 +200,7 @@ begin
         Application.Initialize;
         Application.Title := 'Keyman Configuration';
         Application.CreateForm(TmodWebHttpServer, modWebHttpServer);
-        try
+  try
```
Comment (ermshiperete) [inline windows/src/desktop/kmshell/kmshell.dpr:203]:
```suggestion
        try
```

**Review by ermshiperete — APPROVED** [review]
LGTM.

> Note: the 'SKUpdate_...' text will be replaced with the actual update description in a full build.

I assume these lines in the table are keyboards that have an update available?

File: windows/src/desktop/kmshell/main/UfrmMain.pas [hunk 2]
```diff
@@ -345,6 +348,9 @@ procedure TfrmMain.FireCommand(const command: WideString; params: TStringList);
   else if command = 'support_updatecheck' then Support_UpdateCheck
   else if command = 'support_proxyconfig' then Support_ProxyConfig
 
+  else if command = 'update_checknow' then Update_CheckNow
+
+
```
Comment (rc-swag) [inline windows/src/desktop/kmshell/main/UfrmMain.pas:353]:
```suggestion
  else if command = 'update_checknow' then Update_CheckNow

```

**Review by rc-swag — APPROVED** [review]
lgtm

**Comment by mcdurdin** [comment]
> I assume these lines in the table are keyboards that have an update available?

Yes

**Comment by rc-swag** [comment]
Just a note I think the "strings.xml" update must be missing from this check-in. It is needed for display the "Apply updates now" and the "Check for new updates". Also not sure which icon image is used for the windows update. I can add these quickly but thought I might mention it in the off chance @mcdurdin still has the update on his machine.

**Comment by mcdurdin** [comment]
> I can add these quickly but thought I might mention it in the off chance @mcdurdin still has the update on his machine.

I don't think I have them? I'm not 100% sure what you are looking for though?

**Comment by rc-swag** [comment]
> > I can add these quickly but thought I might mention it in the off chance @mcdurdin still has the update on his machine.
> 
> I don't think I have them? I'm not 100% sure what you are looking for though?

No problem I have added them on one of my branches. `S_Button_Update_ApplyNow` and `S_Button_Update_Check` in `keyman_update.xsl` do not have entries in `strings.xml`

**Pull request merged**